### PR TITLE
GOINGTOSAI: SWI-4543 - Prepare saiqosmap.py file for upstreaming

### PR DESCRIPTION
**Pull request by kmalek-priv**

### CODE REVIEW

**Review by ravi861 — APPROVED** [review]
LGTM. Please proceed to upstream

File: ptf/saiqosmap.py [hunk 1]
```diff
@@ -0,0 +1,9596 @@
...
+        self.l3IPv6QosSameDscpToTcMappingManyPortsTest()
+        self.l3IPv4QosVariousDscpToTcMappingManyPortsTest()
+        self.l3IPv6QosVariousDscpToTcMappingManyPortsTest()
+
+    def tearDown(self):
+        super(L3QosDscpToTcTest, self).tearDown()
+
+    def l3IPv4QosVariousDscpToTcMappingManyPortsTest(self):
+        '''
+        L3 IPv4 DSCP to TC mapping for various ingress ports test.
```
Comment (kperumalbfn) [inline ptf/saiqosmap.py]: Please add description that TC-->Queue mapping is also done and how the tests are validated.

Comment (kperumalbfn) [reply ptf/saiqosmap.py]: Also, can you add a table with all the below TCs with qos mappings from DSCP-->TC-->Queue.

Comment (kperumalbfn) [reply ptf/saiqosmap.py]: Please add the table mapping for all the tests

File: ptf/saiqosmap.py [hunk 2]
```diff
@@ -0,0 +1,9596 @@
...
+        '''
+        print("l3IPv4QosVariousDscpToTcMappingManyPortsTest")
```
Comment (kperumalbfn) [inline ptf/saiqosmap.py]: We are doing only routing traffic. Can you also add regular switching traffic for DSCP-->TC-->Queue.

File: ptf/saiqosmap.py [hunk 3]
```diff
@@ -0,0 +1,9596 @@
...
+        time.sleep(8)
+        post_cnt = self.getPortPPGStats(self.test_dev_ports)
+        self.assertEqual(
+            self.comparePortPPGStats(expected_cnt, post_cnt), True)
+        print("\tok")
+
+    def variousPfcToPpgMapOnManyIngressPortsIPv4(self):
+        """
+        Function configures the PFC to Priority Group test data
+        for various pfc -> ppg map on many ingress ports
```
Comment (kperumalbfn) [inline ptf/saiqosmap.py]: Add the description for validation like queue or ppg stats in all the tests.

File: ptf/saiqosmap.py [hunk 4]
```diff
@@ -0,0 +1,9596 @@
...
+        # get the initial counters
+        expected_cnt = self.getPortPPGStats(self.test_dev_ports)
+        for port in range(0, len(self.test_dev_ports)):
+            ingress_port = self.test_dev_ports[port]
+            for key, ppg in self.test_mapping[port]:
+                ingress_ip_tos = key << 2
+                pkts = 1 + port
+                expected_cnt[port][ppg] += pkts
+                self.ipv6_pkt.tc = ingress_ip_tos
+                send_packet(self, ingress_port, self.ipv6_pkt, pkts)
```
Comment (kperumalbfn) [inline ptf/saiqosmap.py]: Can you add verify packet in all the missing cases.

File: ptf/saiqosmap.py [hunk 5]
```diff
@@ -0,0 +1,9596 @@
...
+            self.assertEqual(
+                update_and_verify_qos_map(
+                    self.client, pfc_to_pg_map_id,
+                    SAI_QOS_MAP_TYPE_PFC_PRIORITY_TO_PRIORITY_GROUP, pfc_list,
+                    pfc_pg_list), True,
+                "Failed verify updated qos_map attribute")
+            # update QOS MAP TC TO PRIORITY GROUP
+            self.assertEqual(
+                update_and_verify_qos_map(
+                    self.client, tc_to_pg_map_i
```
Comment (kperumalbfn) [inline ptf/saiqosmap.py]: Add switching traffic test cases in addition to routing

Comment (kperumalbfn) [inline ptf/saiqosmap.py]: Are we doing queue mapping in this test? please update

Comment (kperumalbfn) [inline ptf/saiqosmap.py]: There are no packet tests for the modify cases, please add them

**Pull request closed without merging**